### ISSUE TITLE
Pattern matching issues

### ISSUE THREAD

**Issue by VBAndCs** [paul1956/CSharpToVB#83]
Please look at this:
```C#
		public override bool Equals(object obj)
		{
			if (obj is Span span)
			{
				if (span.start == start)
				{
					return span.length == length;
				}
				return false;
			}
			return false;
		}
```

It is converted to:
```VB
' To configure or remove Option's included in result, go to Options/Advanced Options...
Option Compare Text
Option Explicit On
Option Infer Off
Option Strict On

Public Overrides Function Equals(obj As Object) As Boolean
    Dim TempVar As Boolean = TypeOf obj Is Span1
    Dim span As Span1 = obj
    If TempVar Then
        If span.start = start Then
            Return span.length = length
        End If
        Return False
    End If
    Return False
End Function

```

Which can cause runtime errors.
It should be:
```vb
Public Overrides Function Equals(obj As Object) As Boolean
    If TypeOf obj Is Span Then
        Dim span = Ctype(obj, Span)
        If span.start = start Then
            Return span.length = length
        End If
        Return False
    End If
    Return False
End Function

```

And why to change the type span to Span1?

**Comment by paul1956** [comment]
@VBAndCs span to Span1 is a separate bug that I don't know when I will be able to fix. The real issue is the Dim span is in the wrong place it needs to be after the second If. I will look at that.

**Comment by VBAndCs** [comment]
I never saw this Span1 before the last version. It always was:
`Dim Span1 as Span`
but this is new:
`Dim Span as Span1`

**Comment by VBAndCs** [comment]
And look at this too:
```c#
		public override bool Equals(object obj)
		{
			if (obj is String s)
			{

			}
			return false;
		}
```

It gives:
```vb
Public Overrides Function Equals(obj As Object) As Boolean
    Dim TempVar As Boolean = TypeOf obj Is String
    Dim s As String
    String.TryParse(obj.ToString, s)

    If TempVar Then
            
    End If
    Return False
End Function
```

the `    String.TryParse(obj.ToString, s)` location is wrong also, and String doesn't contain a TryParse, besides there is no need to try as the type is already checked. A Ctype will do just fine inside the If block

**Comment by paul1956** [comment]
@VBAndCs both issues are fixed in converter 5.0.2.11
